### PR TITLE
Register plug-in as Packer integration

### PR DESCRIPTION
**Pull request by nywilken**
:wave: fellow Packer maintainer working on migrating your plugin to the Packer integration framework.

This change takes the necessary steps to register this plugin as an official Packer integration.
Integrations can be found on the Packer integration portal at https://developer.hashicorp.com/packer/integrations.

---

The pull-request consists of the following changes:

- Adds controlling, metadata file, `metadata.hcl` for registering the plug-in and it components as integrations.
Details on the contents, along with a description of the attributes, can be found at https://github.com/hashicorp/integration-template.
- Adds the GitHub action workflows for triggering manual and automatic integration updates.
- Restructures the plug-in documentation to match the expected format of the integration framework.
- Adds a `.web-docs` directory for serving the fully render documentation as the integration docs.
- Adds the build-docs make target `make build-docs` for syncing changes to the `docs` directory to the `.web-docs` directory.

Changes to the integration docs can be made at plugin release via the `notify-integration-release-via-tag` workflow or
manually by running the `notify-integration-release-via-manual` workflow.

Details on how the Integration framework pipeline works can be found at https://github.com/hashicorp/packer/pull/12702

### TODOs
- [x] Open pull-request against external plugin.
- [x] Update integration description [`.web-docs/metadata.hcl`](.web-docs/metadata.hcl).
- [x] Packer team open internal pull-request to enable integration.
- [ ] Review plugin integration on Packer integration portal .... Iterate

### CODE REVIEW

**Comment by chris-rock** [comment]
@nywilken @BrandonRomano Thank you for the help to move it to the packer integration portal. Once @BrandonRomano agreed to the CLA, we can merge it and take care of the spelling issues.

**Comment by chris-rock** [comment]
@BrandonRomano is covered by the HashiCorp CLA. @benr can you help turn his CLA to green?

File: .web-docs/metadata.hcl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# For full specification on the configuration of this file visit:
+# https://github.com/hashicorp/integration-template#metadata-configuration
+integration {
+  name = "Mondoo"
+  description = "TODO"
```
Comment (nywilken) [inline .web-docs/metadata.hcl:5]:
```suggestion
  description = "Scans Linux and Windows HashiCorp Packer builds for vulnerabilities and security misconfigurations."
```

**Comment by nywilken** [comment]
> @nywilken @BrandonRomano Thank you for the help to move it to the packer integration portal. Once @BrandonRomano agreed to the CLA, we can merge it and take care of the spelling issues.

Thank you @chris-rock. There are a few small updates that need to be made to complete the PR but please feel free to review the docs when you or the team have time. [lbajolet-hashicorp](https://github.com/lbajolet-hashicorp) will be assisting in getting this ready over the next couple of days.

**Comment by chris-rock** [comment]
recheck

File: .web-docs/components/provisioner/cnspec/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  If `score_threshold` is set to a value, and `on_failure = "continue"`
+  builds will continue regardless of what score is returned.
+
+- `labels` (map[string]string) - Configure an optional map of labels for the asset data in Mondoo Platform.
+
+- `annotations` (map[string]string) - Configure an optional map of `key/val` annotations for the asset data in
+  Mondoo Platform.
+
+- `incognito` (bool) - Configures incognito mode. Defaults to `true`. When set to false, scan results
+  will not be sent to the Mondoo platform.
```
Comment (tas50) [inline .web-docs/components/provisioner/cnspec/README.md:63]:
```suggestion
  will not be sent to the Mondoo Platform.
```

File: .web-docs/components/provisioner/mondoo/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  If `score_threshold` is set to a value, and `on_failure = "continue"`
+  builds will continue regardless of what score is returned.
+
+- `labels` (map[string]string) - Configure an optional map of labels for the asset data in Mondoo Platform.
+
+- `annotations` (map[string]string) - Configure an optional map of `key/val` annotations for the asset data in
+  Mondoo Platform.
+
+- `incognito` (bool) - Configures incognito mode. Defaults to `true`. When set to false, scan results
+  will not be sent to the Mondoo platform.
```
Comment (tas50) [inline .web-docs/components/provisioner/mondoo/README.md:77]:
```suggestion
  will not be sent to the Mondoo Platform.
```

File: .web-docs/components/provisioner/cnspec/README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+- `use_proxy` (bool) - Use proxy to connect to host to scan. This configuration will fall-back to packer proxy
+  for cases where the provisioner cannot access the target directly
+  NOTE: we have seen cases with the vsphere builder
+
+- `output` (string) - Set output format: summary, full, yaml, json, csv, compact, report, junit (default "compact")
+
+- `score_threshold` (int) - An integer value to set the `score_threshold` of mondoo scans. Defaults to `0` which results in
+  a passing score regardless of what scan results are returned.
+
+- `mondoo_config_path` (string) - The path to the mondoo client config. Defaults to `$HOME/.config/mondoo/mondoo.yml`
```
Comment (tas50) [inline .web-docs/components/provisioner/cnspec/README.md:84]:
```suggestion
- `mondoo_config_path` (string) - The path to the Mondoo client config. Defaults to `$HOME/.config/mondoo/mondoo.yml`
```

File: .web-docs/components/provisioner/mondoo/README.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  If `score_threshold` is set to a value, and `on_failure = "continue"`
+  builds will continue regardless of what score is returned.
+
+- `labels` (map[string]string) - Configure an optional map of labels for the asset data in Mondoo Platform.
+
+- `annotations` (map[string]string) - Configure an optional map of `key/val` annotations for the asset data in
+  Mondoo Platform.
+
+- `incognito` (bool) - Configures incognito mode. Defaults to `true`. When set to false, scan results
+  will not be sent to the Mondoo Platform.
```
Comment (tas50) [inline .web-docs/components/provisioner/mondoo/README.md:77]:
```suggestion
  will not be sent to Mondoo Platform.
```

File: provisioner/provisioner.go [hunk 6]
```diff
@@ -88,7 +88,7 @@ type Config struct {
 	// Mondoo Platform.
 	Annotations map[string]string `mapstructure:"annotations"`
 	// Configures incognito mode. Defaults to `true`. When set to false, scan results
-	// will not be sent to the Mondoo platform.
+	// will not be sent to the Mondoo Platform.
```
Comment (tas50) [inline provisioner/provisioner.go:91]:
```suggestion
	// will not be sent to Mondoo Platform.
```

**Comment by chris-rock** [comment]
Happy new year @nywilken @BrandonRomano. Is there anything missing from your side, otherwise we would love to get this merged.

**Comment by nywilken** [comment]
Hi @chris-rock thanks for bubbling this up. We are coming back from Holidays this week so catching up. Happy New Year. This PR is all good to go we enabled the integration on our end so you can merge and trigger the .github/workflows/notify-integration-release-via-manual.yaml job when ready with the existing plugin version.

**Review by chris-rock — APPROVED** [review]
Thank you @nywilken @lbajolet-hashicorp

**Pull request merged**